src/RandomSetGenerator.java

```diff
@@ -0,0 +1,106 @@
+
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+
+public class RandomSetGenerator {
+
+	static private Random randomNumberGenerator;
+	
+	public RandomSetGenerator() {
+		this(new SecureRandom()); 
+	}
+
+	public RandomSetGenerator(Random generator) {
+		randomNumberGenerator = generator;
+	}
+
+	public List<Integer> generate(int quantity, int maxValue) throws IllegalArgumentException {
+		Set<Integer> randomSet;
```

Comment:
make this declaration 

Set<Integer> randomSet = new HashSet<Integer>();

that way, you clean up the if-else conditional below, and you can call .size() to see the current size of the Set.